src/monster.cpp

```diff
@@ -942,6 +942,17 @@ void Monster::onThinkDefense(uint32_t interval)
 				continue;
 			}
 
+			uint32_t summonCount = 0;
+			for (Creature* summon : summons) {
+				if (!summon->isRemoved() && summon->getName() == summonBlock.name) {
```

Comment:
You can remove the isRemoved() check since we use summons.size() to check for maxSummons above.